### PR TITLE
Fix enum formatting again

### PR DESCRIPTION
**Pull request by Paxxi**
## Description

This attempt removes the enums from fmt altogether
as the wrapper functions will convert to int before fmt
sees them.

This makes the changes independent from fmt so we don't
have to worry about different behaviour for different versions.

## How Has This Been Tested?

tests pass
## Types of change

- [x] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [x] I have added tests to cover my change
- [x] All new and existing tests passed

### CODE REVIEW

File: xbmc/utils/StringUtils.h [hunk 1]
```diff
@@ -47,33 +57,16 @@ DEF_TO_STR_VALUE(foo) // outputs "4"
...
-// clang-format off
-// This is required to format enum classes with fmt as they're not implicitly
-// convertible to int.
-// This operator has to be declared before including format.h and ostream.h.
-// We keep clang format disabled to avoid reordering and breaking things
-// https://github.com/fmtlib/fmt/issues/391
-template<typename T, typename std::enable_if_t<std::is_enum<T>::value, int> = 0>
-std::ostream& operator<<(std::ostream& os, const T& value)
+template<typename T, std::enable_if_t<!std::is_enum<T>::value, int> = 0>
+constexpr auto&& enum_to_int(T&& arg) noexcept
```
Comment (Montellese) [inline xbmc/utils/StringUtils.h]: Why does this method not follow our naming conventions?

**Review by Montellese — COMMENTED** [review]
Looks good but I would split the PR into three commits:

1. revert the commit from the previous PR to clearly mark that it didn't solve the problem.
2. clang formatting
3. the proper fix commit with `enum_to_int()`

File: xbmc/utils/StringUtils.h (hunk 1 above)
Comment (Paxxi) [reply xbmc/utils/StringUtils.h]: I got sucked in by the type traits naming :)

**Comment by Paxxi** [comment]
ok cleaned up

**Comment by Paxxi** [comment]
ping @Montellese @MilhouseVH tested it with success, good to merge?

**Review by Montellese — APPROVED** [review]
Sorry I thought I already approved.

**Pull request merged**